src/Microsoft.ML.ImageAnalytics/ImageGrayscaleTransform.cs

```diff
@@ -69,50 +78,57 @@ private static VersionInfo GetVersionInfo()
 
         private const string RegistrationName = "ImageGrayscale";
 
-        // Public constructor corresponding to SignatureDataTransform.
-        public ImageGrayscaleTransform(IHostEnvironment env, Arguments args, IDataView input)
-            : base(env, RegistrationName, env.CheckRef(args, nameof(args)).Column, input, t => t is ImageType ? null : "Expected Image type")
+        public IReadOnlyCollection<(string input, string output)> Columns => ColumnPairs.AsReadOnly();
```

Comment:
>public [](start = 8, length = 6)

internal maybe?

Reply:
On other hand, I kinda want this to be in base class

---
In reply to: [213856192](https://github.com/dotnet/machinelearning/pull/753#discussion_r213856192) [](ancestors = 213856192)

Reply:
which is ColumnPairs... why we need this one?

---
In reply to: [213858620](https://github.com/dotnet/machinelearning/pull/753#discussion_r213858620) [](ancestors = 213858620,213856192)